backwardcompatibilityml/tensorflow/models.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import tensorflow.compat.v2 as tf
+
+
+class BCNewErrorCompatibilityModel(tf.keras.models.Sequential):
+
+    def __init__(self, *args, h1=None, lambda_c=None, **kwargs):
+        """
+        Args:
+          h1: An existing tensorflow model that has been pre-trained,
+            and which we want to be compatible with.
+        lambda_c: A floating point value between 0.0 and 1.0 that is
+          used as a regularization parameter to weight how much the
+          dissonance is used to penalize the loss while training.
+        """
+        super(BCNewErrorCompatibilityModel, self).__init__(*args)
+        self.h1 = h1
+        self.lambda_c = lambda_c
+
+    def dissonance(self, h2_output, target_labels, loss):
+        """
+        The dissonance function, which uses the loss function
+        specified by the user to calculate the loss on a subset
+        of the target.
+        """
+        cross_entropy_loss = loss(target_labels, h2_output)
+        return cross_entropy_loss
+
+    def loss_func(self, x, y, loss=None):
+        """
+        Backward compatibility loss function to be used by the model
+        """
+        if loss is None:
+            loss = tf.keras.losses.SparseCategoricalCrossentropy(
+                reduction=tf.keras.losses.Reduction.SUM)
+        h1_output = tf.argmax(self.h1(x), axis=1)
```

Comment:
This fails if h1=None. Are you sure you want to allow h1=None in the constructor?

Reply:
Good point. Though I can't think of any scenario where the user would not set `h1`.
What I can do is just raise an exception at instantiation time if the user sets it to None. And make sure to document that we need that.